src/main/java/com/github/manolo8/darkbot/backpage/HangarManager.java

```diff
@@ -97,28 +88,92 @@ public void updateHangars() {
         if (hangarData == null) return;
 
         hangars.clear();
-        forEachHangar(hangarData, h -> hangars.add(GSON.fromJson(h, Hangar.class)));
+        forEachRet(hangarData, h -> hangars.add(GSON.fromJson(h, Hangar.class)),"hangars");
+    }
+
+    public String getActiveHangar() {
+        for (Hangar hangar : hangars) {
+            if (hangar.hangarIsActive()) return hangar.getHangarId();
+        }
+        return null;
+    }
+
+    public String getHangarData() {
```

Comment:
You're calling getHangarData from different places, to update different infos.
Since you've already made a (somewhat expensive, time-wise) call, you should update all info you can get from this.

make this a `public void updateHangarData()` wich calls updateDrones, updateItemInfos etc with the result.

You could also add "getOrFetchDrones" etc, wich will get the drones if they've been updated, or fetch them and return them if they haven't.